src/app/switch-builder/switch-builder.component.html

```diff
@@ -0,0 +1,8 @@
+<div [ngSwitch]="type">
+  <div *ngSwitchCase="'button'" ><app-button  [styles]="styles$ | async"></app-button></div>
+  <div *ngSwitchCase="'checkbox'"><app-checkbox [styles]="styles$ | async" ></app-checkbox></div>
+  <div *ngSwitchCase="'input'"><app-input [styles]="styles$ | async"></app-input></div>
+  <div *ngSwitchCase="'label'"><app-label [styles]="styles$ | async"></app-label></div>
+  <div *ngSwitchCase="'select'"><app-select [styles]="styles$ | async"></app-select></div>
+  <div *ngSwitchCase="'textarea'"><app-textarea [styles]="styles$ | async"></app-textarea></div>
```

Comment:
create and use Enum, use ngrxPush and why do we need <div> wrapper?
```suggestion
  <div *ngSwitchCase="Enum.Button" ><app-button  [styles]="styles$ | ngrxPush"></app-button></div>
  <div *ngSwitchCase="'checkbox'"><app-checkbox [styles]="styles$ | async" ></app-checkbox></div>
  <div *ngSwitchCase="'input'"><app-input [styles]="styles$ | async"></app-input></div>
  <div *ngSwitchCase="'label'"><app-label [styles]="styles$ | async"></app-label></div>
  <div *ngSwitchCase="'select'"><app-select [styles]="styles$ | async"></app-select></div>
  <div *ngSwitchCase="'textarea'"><app-textarea [styles]="styles$ | async"></app-textarea></div>
```